#166790055 user should be able to comment on article

pngabo opened the pull request:
#### What does this PR do?
 - comment on article
#### Description 
 - create comment article
 - update comment
 - delete comment
 - get comment
 - get all comments
#### How to test
 - clone the repo => cd (name of repo) => npm install => npm run dev
#### Relevant pivotal tracker story
 -  [#166790055](https://www.pivotaltracker.com/story/show/166790055)
#### screenshots
![Ao2gHmrWa0](https://user-images.githubusercontent.com/31598432/64853529-d9ea9000-d61b-11e9-9d10-2b0aa5ad4828.gif)

kagabof commented:
Great work on this PR @patrickngabo but you can change the commit message, it should not be `[started #166790055]`, it should like `[Finishes #166790055]`

shaluchandwani reviewed (dismissed):
Good work on the PR @patrickngabo, although I found some issues Please address them 
1.when you are showing success message the window reloads very fast that we are not able to see the message, just add a delay of some time like "setTimeout(() => window.location.reload(), 6000))"

2.while handling errors the comments are not visible when the window reloads. please do the same thing there also.

3. In the header dropdown, the username CSS is breaking please check also in the profile page username is breaking.

4. When I comment I am getting the username of the user who created the article on the comment. but I am logged in with another user. Please fix that. The output should have the username and the image of the user. It will be better if that has the date also.

kagabof commented on src/components/comments/CommentComponent.js:
```diff
@@ -0,0 +1,32 @@
+import authorImage from '../../assets/images/profile_plaholder.png';
+
+export class CommentComponent extends Component {
+  render() {
```
@patrickngabo, we agreed that we have to use functional components and pass props from the parent component.
but I can see that in this file you are not following it why?

kagabof commented on src/redux/reducers/getArticlesReducer.js:
```diff
@@ -1,7 +1,7 @@
 
 export const initialState = {
-  articles: [],
+  articles: {},
```
We are using one Initial state but  I can see in this file you are not importing it why?

pngabo replied:
this one is used for articles not for comments

pngabo commented:
> The commit message should not be `[started #166790055]`, it should like `[Finishes #166790055]`
@kagabof I have fixed this one

pngabo commented:
> Good work on the PR @patrickngabo, although I found some issues Please address them
> 1.when you are showing success message the window reloads very fast that we are not able to see the message, just add a delay of some time like "setTimeout(() => window.location.reload(), 6000))"
> 
> 2.while handling errors the comments are not visible when the window reloads. please do the same thing there also.
> 
> 1. In the header dropdown, the username CSS is breaking please check also in the profile page username is breaking.
> 2. When I comment I am getting the username of the user who created the article on the comment. but I am logged in with another user. Please fix that. The output should have the username and the image of the user. It will be better if that has the date also.

I have fixed all those changes now @shaluchandwani

Joe-Joseph reviewed (commented):
Good work on your PR @patrickngabo, kindly change the button color to look like other buttons also don't display the comment card before the article body get displayed.

Commit:
ft(comments): user should be able to comment on article

- create comment article
 - update comment
 - delete comment
 - get comment
 - get all comments

[finishes #166790055]
